src/plonk_util.rs

```diff
@@ -77,6 +74,44 @@ pub(crate) fn halo_n<C: HaloCurve>(s_bits: &[bool]) -> C::ScalarField {
     a * C::ZETA_SCALAR + b
 }
 
+/// Compute `n(x)` for a given `x`, where `n` is the injective function related to the Halo
+/// endomorphism.
+pub(crate) fn halo_n_mul<C: HaloCurve>(s_bits: &[bool], p: AffinePoint<C>) -> AffinePoint<C> {
```

Comment:
nit: maybe call it halo_endo_mul? Just seems easy to forget what `n` refers to in Halo

Reply:
On second thought, since we do have a `halo_n` method, maybe it's good to match that. Up to you.

Reply:
I'll think I'll keep `halo_n_mul` for now to be coherent with the `halo_n` function.